### PR TITLE
Implement -k "keep going" flag

### PR DESCRIPTION
**Pull request by jamesr**
This option tells the build to keep going until N failures are detected or the
build is interrupted.

### CODE REVIEW

File: src/task.rs [hunk 1]
```diff
@@ -220,10 +236,14 @@ fn run_command(cmdline: &str) -> anyhow::Result<TaskResult> {
     // TODO: Set up pipes so that we can print the process's output.
     //output.append(&mut cmd.stdout);
     //output.append(&mut cmd.stderr);
-    let success = exit_code == 0;
+    let termination = match exit_code {
+        0 => Termination::Success,
+        0xC000013A => Termination::Interrupted,
```
Comment (evmar) [inline src/task.rs:241]: O_O

File: src/work.rs [hunk 2]
```diff
@@ -322,6 +324,7 @@ impl<'a> Work<'a> {
             graph,
             db,
             progress,
+            keep_going: keep_going,
```
Comment (evmar) [inline src/work.rs:327]: FWIW you can write this `keep_going,`

**Comment by evmar** [comment]
This looks great, can you resolve the conflicts?  (Sorry, I moved all of main.rs to run.rs...)

**Comment by evmar** [comment]
I rebased this for you and merged it, thanks!

**Comment by evmar** [comment]
Needed to follow this up with:
https://github.com/evmar/n2/commit/8a8013a2eafa958f512259859eea063491970e80

**Comment by jamesr** [comment]
> Needed to follow this up with: [8a8013a](https://github.com/evmar/n2/commit/8a8013a2eafa958f512259859eea063491970e80)

Ah sorry about that - I completely missed that tracking.

**Pull request closed without merging**